.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,233 @@
+            )" |
+            jq -r '.MetricWidgetImage' |
+            base64 -d > soak-tests/snapshots/${{ github.sha }}/${{ matrix.app-platform }}-${{ matrix.instrumentation-type }}-$metric_kind-results-$GITHUB_RUN_ID.png;
+          done
+      - name: Commit snapshots to repository
+        uses: EndBug/add-and-commit@v7
+        with:
+          add: soak-tests/snapshots
+          branch: gh-pages
+          message: 'Adding Soak Tests Snapshots'
+      - name: Move back to the main branch
+        run: git checkout main
+      - name: Prepare Soak Test results as JSON output
+        run: >-
+          declare -a COMMON_API_PARAMETERS=(
+          --start-time $(date -u -d '${{ env.SOAK_TEST_DURATION_IN_MINUTES }} minutes ago' +%FT%TZ)
+          --end-time $(date -u +%FT%TZ)
+          );
+          METRIC_DATA_QUERY_INPUT_JSON=$(
+            jq -n '[ inputs ] | add' \
+            ./.github/aws-cloudwatch-api-values/get-metric-data/prepared-cpu-load-metric-data-query.json \
+            ./.github/aws-cloudwatch-api-values/get-metric-data/prepared-total-memory-metric-data-query.json
+          );
+          aws cloudwatch get-metric-data "${COMMON_API_PARAMETERS[@]}" --cli-input-json "{ \"MetricDataQueries\": $METRIC_DATA_QUERY_INPUT_JSON }" |
+          jq "{
+            benchmarks: [
+              {
+                Name: \"Soak Test Average CPU Load\",
+                Value: first(.MetricDataResults[] | select(.Id == \"cpu_load_expr\")) | .Values | ( add / length / ${{ env.HOSTMETRICS_COLLECTION_INTERVAL_IN_SECONDS }} / $NUM_OF_CPUS * 100 ),
+                Unit: \"Percent\"
+              },
+              {
+                Name: \"Soak Test Average Virtual Memory Used\",
+                Value: first( .MetricDataResults[] | select(.Id == \"total_memory_expr\") ) | .Values | ( add / length / $(echo '2^20' | bc) ),
+                Unit: \"Megabytes\"
+              }
+            ]
+          }" |
+          tee output.json
+      - name: Report on Soak Test Averages results
+        uses: NathanielRN/github-action-benchmark@v1.8.2-alpha3
+        continue-on-error: true
+        id: check-failure-after-soak-tests
+        with:
+          name: Soak Test Results - sample-app-${{ matrix.app-platform }}-${{ matrix.instrumentation-type }}
+          tool: custombenchmark
+          output-file-path: output.json
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          max-items-in-chart: 100
+          alert-threshold: 175%
+          # Does not work as expected, see: https://github.com/open-telemetry/opentelemetry-python/pull/1478
+          # comment-always: true
+          fail-on-alert: true
+          auto-push: ${{ github.ref == 'refs/heads/main' }}
+          gh-pages-branch: gh-pages
+          benchmark-data-dir-path: soak-tests/per-commit-overall-results
+      - name: Publish Issue if failed DURING Soak Tests
+        uses: JasonEtco/create-an-issue@v2
```

Comment:
For aws-observability repos in particular we'd like to avoid third party (not company maintained like GitHub, AWS, Docker) actions if possible. They're fine actions, but a bit less risk.

`gh` CLI is preinstalled what do you think about using it?

https://cli.github.com/manual/gh_issue_create

It will mean some seds I guess.

Reply:
I think the native mustache support in the action is a pretty big feature given the work of reinventing that for ourselves, and this action seems quite widely used. Especially since there are so many substitutions in the current template.

Reply:
I agree that the action seems widely-used and so it would make this step read nicely. We can always reduce the substitutions if we think there are too many but I think we get the most value from these tests and this automatic issue publishing with where it is right now. (We can even add more easily with this action in the future).